adds datetime folder containing DatetimeFeatures nb

dodoarg opened the pull request:
@solegalli 
related to main repo's issue#67

I couldn't really find great datetime datasets on UCI. This one has only one datetime variable which works for most purpose I guess but I can't really show the timezone related functionality for example (which is in the docs examples with the toy dataset though)

Let me know if it would be best to introduce similar toy examples in this notebook too or maybe find a more suitable real dataset looking into other sources. Or if there's something else important I forgot to put in here of course

dodoarg committed:
adds datetime folder containing DatetimeFeatures nb

solegalli commented:
Hi @dodoarg 

Thank you for creating the notebook. Looks really good. I have 3 minor suggestions:

After cell 3, you mention that the transformer **does not support missing data**, but it does, in fact, we have an attribute to control this behavior. Could you change the wording?

In cell 8, the link to the docs (which do not exist already) is not going to be correct. I would remove it, and users can search for the class themselves. We can always add it later after the package is released.

Would you be able to showcase a bit more the attributes of the transformer? in particular the features_to_extract_ when setting the initial param to None or 'all'?

And maybe also show how the variable datetime was captured and stored automatically in variables_ when setting variables=None?

That's all I would say. Then it is pretty much ready to merge.

dodoarg committed:
minor tweaks

dodoarg commented:
> After cell 3, you mention that the transformer does not support missing data, but it does

!! 🤔 yea I removed that 

> In cell 8, the link to the docs is not going to be correct.

It was meant to be some kind of placeholder, I removed it now

> the features_to_extract_ when setting the initial param to None or 'all'

The "all" option was already there, I now added the None/default case in a similar way

> show how the variable datetime was captured and stored automatically in variables_ when setting variables = None

This was actually in the notebook before I decided to remove it - I thought attributes named with a trailing underscore are not meant to be accessed by the user. But that's not how it works, right? 
Anyhow, I put it back in

solegalli commented:
Attributes with a preceding underscore are private. Those will not be visible to the users.

Hi @dodoarg 

Attributes with a trailing underscore, by sklearn convention, contain the values that are learned during fit. In, fact, the function to check that the transformer has been fit (which we use in the tests), checks for the existence of trailing underscores added during fit.

If you have a look at the sklearn classes, you will see that all learned parameters have a trailing underscore, like coef_, feature_importance_, classes_, etc.

Thank you for the great work. I am merging this now.
